sdk/communication/communication-administration/test/utils/recordedClient.ts

```diff
@@ -71,6 +77,45 @@ export function createRecordedCommunicationIdentityClient(
   };
 }
 
+export function createRecordedCommunicationIdentityClientWithToken(
+  context: Context
+): RecordedClient<CommunicationIdentityClient> {
+  const recorder = record(context, environmentSetup);
+
+  if (isBrowser()) {
+    return {
+      client: new CommunicationIdentityClient(env.COMMUNICATION_CONNECTION_STRING),
+      recorder
+    };
+  }
+
+  let credential: TokenCredential;
+
+  if (isPlaybackMode()) {
+    const mockToken = "token";
+    const fakeGetToken = sinon.fake.returns(
+      Promise.resolve({ token: mockToken, expiresOn: new Date() })
+    );
+
+    credential = { getToken: fakeGetToken };
+  } else {
+    credential = new ClientSecretCredential(
+      env.AZURE_TENANT_ID,
+      env.AZURE_CLIENT_ID,
+      env.AZURE_CLIENT_SECRET
+    );
+  }
+
+  return {
+    client: new CommunicationIdentityClient(env.COMMUNICATION_ENDPOINT, credential),
+    recorder
+  };
+}
+
+export function isBrowser() {
```

Comment:
Why not use `!isNode` instead?

Reply:
Using different approach so this has been removed